server/protocols/webadmin-integration-test/webadmin-integration-test-common/src/test/java/org/apache/james/webadmin/integration/vault/DeletedMessageVaultIntegrationTest.java

```diff
@@ -165,25 +159,17 @@ public void setup() throws Throwable {
             .config(WebAdminUtils.defaultConfig()
                 .paramConfig(new ParamConfig(REPLACE, REPLACE, REPLACE)));
     }
-    
-    private GuiceJamesServer createJmapServer(FileSystem fileSystem, Clock clock) throws IOException {
-        return rule.jmapServer(cassandra.getModule(),
-            new TestDeleteMessageVaultPreDeletionHookModule(),
-            binder -> binder.bind(WebAdminConfiguration.class).toInstance(WebAdminConfiguration.TEST_CONFIGURATION),
-            binder -> binder.bind(FileSystem.class).toInstance(fileSystem),
-            binder -> binder.bind(Clock.class).toInstance(clock));
-    }
 
-    private void awaitSearchUpToDate() {
-        rule.await();
-    }
-    
+    protected abstract GuiceJamesServer createJmapServer(FileSystem fileSystem, Clock clock) throws IOException;
+
+    protected abstract void awaitSearchUpToDate();
+
     @After
     public void tearDown() throws Exception {
         jmapServer.stop();
     }
 
-    
+    @Category(BasicFeature.class)
```

Comment:
Please annotate the test class with `@Category(BasicFeature.class)` when you did put `    @Category(BasicFeature.class)` on all methods, this will help readability...